### PR TITLE
Improved BGP docs on MD5

### PR DESCRIPTION
**Pull request by nvibert**
```release-note
Minor documentation fixes and improvements for the BGP MD5 feature
```

### CODE REVIEW

**Review by YutaroHayakawa — APPROVED** [review]
LGTM, thanks!

File: Documentation/network/bgp-control-plane.rst [hunk 1]
```diff
@@ -326,6 +327,13 @@ Cilium Agent's logs rather than a more specific error message.
 
 .. _RFC-2385 : https://www.rfc-editor.org/rfc/rfc2385.html
 
+If a ``CiliumBGPPeeringPolicy`` is deployed with an ``authSecretRef`` that cannot be 
+found, the BGP session will be using an empty password  and an error such as the following will be logged:
+
+::
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]:
```suggestion
If a ``CiliumBGPPeeringPolicy`` is deployed with an ``authSecretRef`` that Cilium cannot find, the BGP session will use an empty password and the agent will log an error such as in the following example::
```

Not sure we need to specify the error itself, it's probably enough to mention there will be an error (the error message is descriptive enough that people should be able to match it with the case you describe), but I'm not opposed to it if you prefer to keep it like this.

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks! Looks good overall, with some suggestions on the phrasing.

File: Documentation/network/bgp-control-plane.rst (hunk 1 above)
Comment (nvibert) [reply Documentation/network/bgp-control-plane.rst]: Thanks @qmonnet - I think folks sometimes forget to look into the agent's log so I think it's worth reminding them and showing them the type of logs that can be on display.

**Review by qmonnet — APPROVED** [review]
Thanks! Please squash the two commits together.

**Review by rastislavs — APPROVED** [review]
Thanks!

**Comment by mhofstetter** [comment]
Adding `ready-to-merge`: all approvals are in and docs change only (`/tests` didn't skipped `ci-ipsec-e2e` properly)

**Pull request merged**